Show events on map

diogo-botelho opened the pull request:
Add method to query database for all events

Function to make a request to backend for an array of existing events

Add logic to Location Marker to make it stationary when not being edited/created:
- Created new component EventMarker, separate from LocationMarker component

Create pins on map for each event

Ensure we don't need to query the backend every time the map loads:
- Instead of altering the state of needEvents, which calls the backend API, we change the events state directly with the new event.

samau3 committed:
Add button that conditionally displays Add Event Form

samau3 committed:
Start Add Event Form

diogo-botelho committed:
Resolved merge conflicts

diogo-botelho committed:
Updated getAll() method in Event Model.

diogo-botelho committed:
Changed the getAll() method in Event Model.

It no longer sends the event id to the frontend.
Fixed the seed data for Events.

diogo-botelho committed:
Added a new Component EventMarker.

diogo-botelho committed:
Added functionality to show existing events in the map.

samau3 commented on frontend/src/EventMarker.tsx:
```diff
@@ -0,0 +1,33 @@
+
+interface LocationMarkerParams {
+  key: string,
+  location:Coordinates
```
Format document to add standard spacing

samau3 commented on frontend/src/EventMarker.tsx:
```diff
@@ -0,0 +1,33 @@
+
+
+export function EventMarker({key, location}:LocationMarkerParams) {
+  // let coordinates = {
+  //   lat: location.lat,
+  //   lng: location.lng
+  // };
```
Remove unnecessary code lines

samau3 commented on frontend/src/EventMarker.tsx:
```diff
@@ -0,0 +1,33 @@
+  const lng = location.lng;
+
+  const position = [];
+  if(lat&&lng) {
```
Format document

samau3 commented on frontend/src/Home.tsx:
```diff
@@ -9,6 +9,17 @@ import { AddEventFormData } from "./interfaces/AddEventFormData";
+  events: Event[];
+}
+
+interface Event {
+  buskerId: number;
+  title: string,
+  type: string,
+  coordinates: Coordinates;
+};
+
```
Move reused interface into a export interface file

samau3 commented on frontend/src/Home.tsx:
```diff
@@ -22,6 +33,29 @@ function Home() {
   // have a state that checks if user is an artist to conditionally show add event button
   const [isAddingEvent, setIsAddingEvent] = useState(false);
   const [coordinates, setCoordinates] = useState<Coordinates | undefined>(undefined);
+  const [events, setEvents] = useState<Event[]>([]);
```
Format document

samau3 commented on frontend/src/Home.tsx:
```diff
@@ -47,19 +81,26 @@ function Home() {
     console.log ("Please select a location");
     } else {
       await BuskApi.createEvent(eventDetails);
+      setEvents(previousData => ([
+        ...previousData,
+        eventDetails
+      ]))
```
Make this into a single line instead? e.g. `setEvents(prevData => [...prevData, eventDetails])`

samau3 commented on frontend/src/LocationMarker.tsx:
```diff
@@ -9,6 +9,10 @@ interface LocationContextType {
   updateCoordinates: (coordinates: Coordinates) => void;
 }
 
+interface LocationMarkerParams {
+  key: string|null,
+  location:Coordinates|undefined
+}
```
Format document

samau3 commented on frontend/src/api.ts:
```diff
@@ -76,6 +76,15 @@ class BuskApi {
+  /**
+   * function to get all events. Returns [event, event,event] */
+  static async getEvents() {
+    // console.log("api call", eventDetails);
```
remove unused console.log

diogo-botelho committed:
Addressed all pull request comments.

Formatted all files.
Moved reused interfaces into own files + added imports in respective files.
Removed unused code.
